vendor/bat-native-ledger/src/bat/ledger/internal/legacy/client_properties.cc

```diff
@@ -47,4 +68,133 @@ bool ClientProperties::operator!=(
   return !(*this == rhs);
 }
 
+base::Value::Dict ClientProperties::ToValue() const {
+  base::Value::Dict dict;
+  dict.Set(kWalletInfoKey, wallet_info.ToValue());
+  dict.Set(kBootTimestampKey, base::Int64ToValue(boot_timestamp));
+  dict.Set(kReconcileTimestampKey, base::Int64ToValue(reconcile_timestamp));
+  dict.Set(kFeeAmountKey, fee_amount);
+  dict.Set(kUserChangedFeeKey, user_changed_fee);
+  dict.Set(kRewardsEnabledKey, rewards_enabled);
+  dict.Set(kAutoContributeKey, auto_contribute);
+
+  base::Value::Dict inline_tips_dict;
+  for (const auto& [key, value] : inline_tips) {
+    inline_tips_dict.Set(key, value);
+  }
+  dict.Set(kInlineTipsKey, std::move(inline_tips_dict));
+
+  return dict;
+}
+
+bool ClientProperties::FromValue(const base::Value::Dict& dict) {
+  // Wallet Info
+  const auto* wallet_info_dict = dict.FindDict(kWalletInfoKey);
+  if (!wallet_info_dict) {
+    NOTREACHED();
+    return false;
+  }
+  if (!wallet_info.FromValue(*wallet_info_dict)) {
+    NOTREACHED();
+    return false;
+  }
+
+  // Boot Timestamp (There is no support for uint64_t. Writing JSON with such
+  // types violates the spec. As we need a uint64_t, we need to use an unsigned
+  // int and cast to a uint64_t)
```

Comment:
I know it's not yours, but this comment is a bit vague - e.g. which spec? Also, in "we need to use an `unsigned int`", shouldn't that be `int64_t`?

Reply:
The spec in question is the JS lang spec. Regarding the type, rapidjson was serialising the value as a `int64_t`, which is in violation of the JS spec, and then the existing code is reading this value as a double, and casting it back to `int64_t`. This operation is not allowed with `base::Value`, so I changed it to have the `int64_t` saved as a string, and then reversed back to string. The previous path for the double conversion is still there though for backwards compatibility/migration.

Reply:
Can we not use `base::ValueToTime` and `TimeToValue` or their new equivalents

Reply:
Fixed in 07e32b475489a39e35b2bc34d9e3cdc57bdaf4cf.